### DISCUSSION TITLE
How to import custom pytestmark? (analogy with pytest_plugins)

### DISCUSSION THREAD

**Discussion by vamotest** [pytest-dev/pytest#10188] [Q&A] [answered]
Hi!

Please tell me how to import [pytestmark](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html#pytest-mark-parametrize) with parameterization correctly?

Conditionally, I can have a file:
```python3
# tests_helpers/mark/some_mark.py
pytestmark = pytest.mark.parametrize("n,expected", [(1, 2), (3, 4)])
```

And I will make an import in my test
```python
from tests_helpers.mark.some_mark import pytestmark
```

Yes, this will work conditionally, but the import will be highlighted in the IDE as unused.

Maybe there is a mechanism for connecting your fixture to conftest?
```python
pytest_plugins = [
    'tests_helpers.pytest_fixtures.unit.commands',
]
```

<img width="915" alt="Снимок экрана 2022-08-04 в 23 26 33" src="https://user-images.githubusercontent.com/20281995/182946109-f4c11fa1-c12e-4cff-ab1b-174cc86ca7ae.png">

**Comment by nicoddemus** [comment, marked as answer]
Hi,

Importing the mark like you did is the way to go. Most IDEs have a mechanism to make it not consider the import unused, usually by adding a comment to it.

**Reply by vamotest** [reply]
Thank you so much for the answer. 
I know about
```python3
from tests_helpers.mark.some_mark import pytestmark  # noqa
```
but I thought there was a more elegant option

**Reply by nicoddemus** [reply]
Given that you want to reuse it, perhaps you can give it a more descriptive name, say:

```python
parametrize_for_numbers = pytest.mark.parametrize(...)
```

And then in your module:

```python
from test_helpers.mark.some_marks import parametrize_for_numbers 

pytestmark = parametrize_for_numbers 
```

This way you avoid the IDE warning and also can have multiple "parametrization marks" for reuse in your test_helpers.

**Reply by vamotest** [reply]
Thank you
